bridge/contracts/AllowTokens/AllowTokens.sol

```diff
@@ -12,192 +12,213 @@ import "../zeppelin/upgradable/ownership/UpgradableSecondary.sol";
+		require(isTokenAllowed(chainId, token), "AllowTokens: Not whitelisted");
+		require(amount >= limit.min, "AllowTokens: Lower than limit");
+
+		// solium-disable-next-line security/no-block-members
+		if (block.timestamp > info.lastDay + 24 hours) { // solhint-disable-line not-rely-on-time
+			// solium-disable-next-line security/no-block-members
+			info.lastDay = block.timestamp; // solhint-disable-line not-rely-on-time
+			info.spentToday = 0;
+		}
+		uint maxWithdraw = _calcMaxWithdraw(info, limit);
+		require(amount <= maxWithdraw, "AllowTokens: Exceeded limit");
+		info.spentToday = info.spentToday.add(amount);
+		setTokenInfoByTokenAddress(chainId, token, info);
+
+		emit UpdateTokensTransfered(token, info.lastDay, info.spentToday);
+	}
+
+	function _addTokenType(string memory description, Limits memory limits) private returns(uint256 len) {
+		require(bytes(description).length > 0, "AllowTokens: Empty description");
+		len = typeDescriptions.length;
+		require(len + 1 <= MAX_TYPES, "AllowTokens: Reached MAX_TYPES");
+		typeDescriptions.push(description);
+		_setTypeLimits(len, limits);
+		emit TokenTypeAdded(len, description);
+		return len;
+	}
+
+	function addTokenType(string calldata description, Limits calldata limits) external onlyOwner returns(uint256 len) {
+		return _addTokenType(description, limits);
+	}
+
+	function _setTypeLimits(uint256 typeId, Limits memory limits) private {
+		require(typeId < typeDescriptions.length, "AllowTokens: bigger than typeDescriptions");
+		require(limits.max >= limits.min, "AllowTokens: maxTokens smaller than minTokens");
+		require(limits.daily >= limits.max, "AllowTokens: dailyLimit smaller than maxTokens");
+		require(limits.mediumAmount > limits.min, "AllowTokens: limits.mediumAmount smaller than min");
+		require(limits.largeAmount > limits.mediumAmount, "AllowTokens: limits.largeAmount smaller than mediumAmount");
+		typeLimits[typeId] = limits;
+		emit TypeLimitsChanged(typeId, limits);
+	}
+
+	function setTypeLimits(uint256 typeId, Limits memory limits) public onlyOwner {
+		_setTypeLimits(typeId, limits);
+	}
+
+	function getTypesLimits() external view override returns(Limits[] memory limits) {
+		limits = new Limits[](typeDescriptions.length);
+		for (uint256 i = 0; i < typeDescriptions.length; i++) {
+			limits[i] = typeLimits[i];
+		}
+		return limits;
+	}
+
+	function getTypeDescriptionsLength() external view override returns(uint256) {
+		return typeDescriptions.length;
+	}
+
+	function getTypeDescriptions() external view override returns(string[] memory descriptions) {
+		descriptions = new string[](typeDescriptions.length);
+		for (uint256 i = 0; i < typeDescriptions.length; i++) {
+			descriptions[i] = typeDescriptions[i];
+		}
+		return descriptions;
+	}
+
+	function isTokenAllowed(uint256 chainId, address token) public view notNull(token) override returns (bool) {
+		return tokenInfo(chainId, token).allowed;
```

Comment:
If it passes like an chainId or token address that doesn't exists for the `tokenInfo` when accessing the `.allowed` it will return the default value for that type, in this case the type of allowed property is boolean and the default is false